compiler/modules/types/cell.bal

```diff
@@ -0,0 +1,151 @@
+// Implementation specific to basic type cell.
+
+public const CELL_MUT_NONE = 0;
+public const CELL_MUT_LIMITED = 1;
+public const CELL_MUT_UNLIMITED = 2;
+
+public type CellMutability CELL_MUT_NONE|CELL_MUT_LIMITED|CELL_MUT_UNLIMITED;
+
+public type CellAtomicType readonly & record {|
+    SemType t;
```

Comment:
I suggest we maintain the t field in a slightly normalized form, specifically if t is equivalent to TOP or NEVER, it should be represented by TOP or NEVER exactly.  Then instead of having to do isEmpty on t, we just do isNever (which is much cheaper). Similarly instead of doing isSameType(t, TOP) we can simply do `t === TOP`.  We do a similar thing in list/mapping, but just for NEVER.

Reply:
if we consider isNever() logic vs isEmpty() logic, they both do
```bal
if t is UniformTypeBitSet {
        return t == 0;
} 
```
at the very beginning. So could there be a difference? 

Also, in `isSameType(t, TOP)` we do `t1 == t2` at the beginning. Doesn't it simply do the job?

Reply:
@jclark if I got [this](https://github.com/ballerina-platform/nballerina/pull/1052/files#r938425308) right, you are suggesting to do the `isEmpty()`, `isSameType(t, TOP)` check upfront and do the normalization when creating the `CellAtomicType`. Isn't that more expensive than doing `isEmpty()` once when we have the combinedCell? In the combined cell, types can be canceled out when taking the intersection and would have a more simplified `semType` that we can do `isEmpty()` on?

Reply:
Resolving as per the offline discussion.